src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+                return True
+
+            if self.mesh.is_convex and other.mesh.is_convex:
+                # Collision manager covers all cases for convex shapes,
+                # so we can just return the value.
+                return surface_collision
+
+            # PASS 4
+            # Compute intersection and check if it's empty. Expensive but guaranteed
+            # to give the right answer.
+            return not isinstance(self.intersect(other), EmptyRegion)
+
+        if isinstance(other, MeshSurfaceRegion):
+            # PASS 1
+            # Check if bounding boxes intersect. If not, volumes cannot intersect.
+            # For bounding boxes to intersect there must be overlap of the bounds
+            # in all 3 dimensions.
+            range_overlaps = [(self.mesh.bounds[0,dim] <= other.mesh.bounds[1,dim]) and \
+                              (other.mesh.bounds[0,dim] <= self.mesh.bounds[1,dim]) \
+                              for dim in range(3)]
+            bb_overlap = all(range_overlaps)
+
+            if not bb_overlap:
+                return False
+
+            # PASS 2
+            # Use Trimesh's collision manager to check for intersection.
+            # If the surfaces collide (or surface is contained in the mesh), 
+            # that implies a collision of the volumes. Cheaper than computing 
+            # intersection. Must use a SurfaceCollisionTrimesh object for the surface 
+            # mesh to ensure that a collision implies surfaces touching.
+            collision_manager = trimesh.collision.CollisionManager()
+
+            collision_manager.add_object("SelfRegion", self.mesh)
+            collision_manager.add_object("OtherRegion", SurfaceCollisionTrimesh(faces=other.mesh.faces, vertices=other.mesh.vertices))
+
+            surface_collision = collision_manager.in_collision_internal()
+
+            if surface_collision:
+                return True
+
+            # PASS 3
+            # Compute intersection and check if it's empty. Expensive but guaranteed
+            # to give the right answer.
+            return not isinstance(self.intersect(other), EmptyRegion)
+
+        if isinstance(other, PolygonalFootprintRegion):
+            # Determine the mesh's vertical bounds (adding a little extra to avoid mesh errors) and
+            # the mesh's vertical center.
+            vertical_bounds = (self.mesh.bounds[0][2], self.mesh.bounds[1][2])
+            mesh_height = vertical_bounds[1] - vertical_bounds[0] + 1
+            centerZ = (vertical_bounds[1] + vertical_bounds[0])/2
+
+            # Compute the bounded footprint and recursively compute the intersection
+            bounded_footprint = other.approxBoundFootprint(centerZ, mesh_height)
+
+            return self.intersects(bounded_footprint)
+
+        if not triedReversed:
+            return other.intersects(self)
+
+        raise NotImplementedError(f"Cannot check intersection of MeshRegion with {type(other)}.")
```

Comment:
I think it would be better to replace these last 3 lines with a call to `super()` and improve the error message there to mention the types of the regions (since this behavior will be common across all region types).